src/Common/OpenTelemetryTraceContext.h

```diff
@@ -19,6 +53,94 @@ struct OpenTelemetryTraceContext
     // Parse/compose OpenTelemetry traceparent header.
     bool parseTraceparentHeader(const std::string & traceparent, std::string & error);
     std::string composeTraceparentHeader() const;
+
+    bool isTraceEnabled() const
+    {
+        return trace_id != UUID();
+    }
+};
+
+/// tracing context kept on thread local
+struct OpenTelemetryThreadTraceContext : OpenTelemetryTraceContext
```

Comment:
In constrast to `OpenTelemetrySpanHolder` which starts a span in function scope, this class initializes the context on one thread and clear the context when it's destroyed.

`OpenTelemetrySpanHolder` relies on the context initialized by `OpenTelemetryThreadTraceContext`.

Since they're doing different initializating, abstracting a common class I think would make things complicated